test/e2e/specs/editor/various/multi-block-selection.spec.js

```diff
@@ -0,0 +1,1367 @@
+			.poll( multiBlockSelectionUtils.getSelectedBlocks )
+			.toEqual( [] );
+
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+
+		await page.keyboard.press( 'Backspace' );
+
+		// Expect both paragraphs to be deleted.
+		await expect.poll( editor.getBlocks ).toEqual( [] );
+	} );
+
+	test( 'should select title if the cursor is on title', async ( {
+		page,
+		editor,
+		pageUtils,
+		multiBlockSelectionUtils,
+	} ) => {
+		for ( let i = 1; i <= 2; i += 1 ) {
+			await editor.insertBlock( {
+				name: 'core/paragraph',
+				attributes: { content: `${ i }` },
+			} );
+		}
+
+		await editor.canvas
+			.getByRole( 'textbox', { name: 'Add title' } )
+			.type( 'Post title' );
+
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+
+		await expect
+			.poll( multiBlockSelectionUtils.getSelectedBlocks )
+			.toEqual( [] );
+		await expect
+			.poll( () =>
+				page.evaluate( () => window.getSelection().toString() )
+			)
+			.toBe( 'Post title' );
+	} );
+
+	test( 'should multi-select in the ListView component with shift + click', async ( {
+		page,
+		editor,
+		multiBlockSelectionUtils,
+		pageUtils,
+	} ) => {
+		for ( let i = 1; i <= 4; i += 1 ) {
+			await editor.insertBlock( {
+				name: 'core/paragraph',
+				attributes: { content: `${ i }` },
+			} );
+		}
+
+		await page
+			.getByRole( 'toolbar', { name: 'Document tools' } )
+			.getByRole( 'button', { name: 'Document Overview' } )
+			.click();
+
+		const listView = page.getByRole( 'treegrid', {
+			name: 'Block navigation structure',
+		} );
+		const navButtons = listView.getByRole( 'gridcell', {
+			name: 'Paragraph link',
+		} );
+
+		await navButtons.nth( 1 ).click();
+		await expect
+			.poll(
+				multiBlockSelectionUtils.getSelectedFlatIndices,
+				'Clicking on the second list item should result in the second block being selected.'
+			)
+			.toEqual( [ 2 ] );
+
+		await navButtons.nth( 3 ).click( { modifiers: [ 'Shift' ] } );
+		await expect
+			.poll(
+				multiBlockSelectionUtils.getSelectedFlatIndices,
+				'Shift clicking the fourth list item should result in blocks 2 through 4 being selected.'
+			)
+			.toEqual( [ 2, 3, 4 ] );
+
+		await navButtons.nth( 0 ).click( { modifiers: [ 'Shift' ] } );
+		await expect
+			.poll(
+				multiBlockSelectionUtils.getSelectedFlatIndices,
+				'With the shift key still held down, clicking the first block should result in the first two blocks being selected.'
+			)
+			.toEqual( [ 1, 2 ] );
+
+		await navButtons.nth( 3 ).click();
+		await expect
+			.poll(
+				multiBlockSelectionUtils.getSelectedFlatIndices,
+				'With the shift key up, clicking the fourth block should result in only that block being selected.'
+			)
+			.toEqual( [ 4 ] );
+
+		// Move focus to the list view link to prepare for the keyboard navigation.
+		await navButtons.nth( 3 ).click();
+		await expect(
+			navButtons.nth( 3 ).getByRole( 'link', { includeHidden: true } )
+		).toBeFocused();
+		// Press Up twice to highlight the second block.
```

Comment:
The original comment suggested that it would "focus" the block, but in reality, it only "highlights" the block.